Tests/scripts/lock_cloud_machines.py

```diff
@@ -0,0 +1,359 @@
+                                                    job_id)
+
+            # We managed to lock a machine
+            if lock_machine_name:
+                lock_machine_list.append(lock_machine_name)
+                number_machines_to_lock -= 1
+                # If we don't need more machines to lock we end the loop
+                if not number_machines_to_lock:
+                    break
+
+                # If the machine was busy we save it to try it again later
+            else:
+                busy_machines.append(machine)
+
+        # Next round we will try and lock only the busy machines
+        list_machines = busy_machines
+
+        # we need more machines but all machines where busy in this round
+        if number_machines_to_lock:
+            sleep(60)
+    return lock_machine_list
+
+
+def create_list_of_machines_to_run(storage_bucket, lock_machine_name, gcs_locks_path, test_machines_path,
+                                   number_machines_to_lock):
+    """
+    get the list of the available machines (or one specific given machine for debugging).
+    Args:
+        storage_bucket(google.cloud.storage.bucket.Bucket): google storage bucket where lock machine is stored.
+        lock_machine_name(str): the name of the file with the list of the test machines.
+        test_machines_path(str): all the exiting machines.
+        gcs_locks_path(str): the lock repository name.
+        number_machines_to_lock(int): the number of the requested machines.
+
+    Returns: the machines list.
+    """
+    if lock_machine_name:  # For debugging: We got a name of a specific machine to use.
+        logging.info('trying to lock the given machine: {options.lock_machine_name}')
+        list_machines = [lock_machine_name]
+    else:
+        logging.info('getting all machine names')  # We are looking for a free machine in all the available machines.
+        test_machines_list = storage_bucket.blob(f'{gcs_locks_path}/{test_machines_path}')
+        list_machines = test_machines_list.download_as_string().decode("utf-8").split()
+        random.shuffle(list_machines)
+
+    if number_machines_to_lock > len(list_machines):
+        logging.error(
+            f'This build requested {number_machines_to_lock} but there are only {len(list_machines)} of active machines')
+        raise
+    return list_machines
+
+
+def wait_for_build_to_be_first_in_queue(storage_client, storage_bucket, gcs_locks_path, job_id, gitlab_status_token):
+    """
+    Run until the job-id is the first in the queue.
+    Args:
+        storage_client(storage.Client): The GCP storage client.
+        storage_bucket(google.cloud.storage.bucket.Bucket): google storage bucket where lock machine is stored.
+        gcs_locks_path(str): the lock repository name.
+        job_id(str): the job id to check.
+        gitlab_status_token(str): the gitlab token.
+    """
+    first_in_the_queue = False
```

Comment:
I think we can add here a random sleep for a couple of seconds before starting to check my number in the queue, in order to prevent 2 builds that were created in the same second to have a raise condition. 
@tkatzir What do you think?

Reply:
Agree.
Something like a random of 1-3 seconds.